code/modules/bitrunning/objects/quantum_server.dm

```diff
@@ -0,0 +1,739 @@
+	var/datum/lazy_template/virtual_domain/to_load
+
+	for(var/datum/lazy_template/virtual_domain/available as anything in subtypesof(/datum/lazy_template/virtual_domain))
+		if(map_key != initial(available.key) || points < initial(available.cost))
+			continue
+		to_load = available
+		break
+
+	if(isnull(to_load))
+		return FALSE
+
+	generated_domain = new to_load()
+	generated_domain.lazy_load()
+
+	return TRUE
+
+/// Loads in necessary map items, sets mutation targets, etc
+/obj/machinery/quantum_server/proc/initialize_map_items()
+	for(var/thing in generated_domain.created_atoms_list)
+		if(isliving(thing)) // so we can mutate them
+			var/mob/living/creature = thing
+			if(creature.can_be_cybercop)
+				mutation_candidates += creature
+
+			continue
+
+		if(istype(thing, /obj/effect/mob_spawn/ghost_role)) // so we get threat alerts
+			RegisterSignal(thing, COMSIG_GHOSTROLE_SPAWNED, PROC_REF(on_threat_created))
+			continue
+
+		if(istype(thing, /obj/effect/mob_spawn/corpse)) // corpses are valid targets too
+			var/obj/effect/mob_spawn/corpse/spawner = thing
+			var/mob/living/creature = spawner.mob_ref?.resolve()
+
+			if(!creature?.can_be_cybercop)
+				continue
+
+			mutation_candidates += creature
+			qdel(spawner)
+
+	var/turf/goal_turfs = list()
+	for(var/obj/thing in generated_safehouse.created_atoms)
+		if(istype(thing, /obj/effect/bitrunning/exit_spawn)) // so there's a place to put the hololadder
+			exit_turfs += get_turf(thing)
+			continue
+
+		if(istype(thing, /obj/effect/bitrunning/goal_turf)) // so there's a place to put the loot
+			var/turf/tile = get_turf(thing)
+			goal_turfs += tile
+			RegisterSignal(tile, COMSIG_ATOM_ENTERED, PROC_REF(on_goal_turf_entered))
+			RegisterSignal(tile, COMSIG_ATOM_EXAMINE, PROC_REF(on_goal_turf_examined))
+			continue
+
+	if(!length(exit_turfs))
+		CRASH("Failed to find exit turfs on generated domain.")
+	if(!length(goal_turfs))
+		CRASH("Failed to find send turfs on generated domain.")
+
+	return TRUE
+
+/// Loads the safehouse
+/obj/machinery/quantum_server/proc/initialize_safehouse()
+	var/datum/turf_reservation/res = generated_domain.reservations[1]
+
+	var/turf/safehouse_load_turf = list()
+	for(var/turf/open/space/space_tile in res.reserved_turfs) // must be a template tile, therefore empty space
+		if(istype(get_area(space_tile), /area/virtual_domain/safehouse/bottom_left))
+			safehouse_load_turf += space_tile
+			break
+
+	if(!length(safehouse_load_turf))
+		CRASH("Failed to find safehouse load landmark on map.")
+
+	var/datum/map_template/safehouse/safehouse = new generated_domain.safehouse_path()
+	safehouse.load(safehouse_load_turf[ONLY_TURF])
+	generated_safehouse = safehouse
+
+	return TRUE
+
+/// Locates any turfs with crate out landmarks
+/obj/machinery/quantum_server/proc/locate_receive_turfs()
+	for(var/turf/open/floor/tile in oview(4, src))
+		if(locate(/obj/effect/bitrunning/reward_spawn) in tile)
+			receive_turfs += tile
```

Comment:
You'll want to make these landmarks and do the same thing with global lists as said above.

Reply:
on it